src/compiler/emitter.ts

```diff
@@ -3770,13 +3937,14 @@ module ts {
                     }
                 }
 
-                function emitAssignmentExpression(root: BinaryExpression) {
-                    var target = root.left;
-                    var value = root.right;
-                    if (root.parent.kind === SyntaxKind.ExpressionStatement) {
-                        emitDestructuringAssignment(target, value);
-                    }
-                    else {
+                function emitAssignmentExpression(root: ExpressionStatement | BinaryExpression) {
+                    // Synthesized nodes will not have parents, so the ExpressionStatements will have to be passed
+                    // in directly. Otherwise, it will crash when we access the parent of a synthesized binary expression.
+                    var emitParenthesized = root.kind !== SyntaxKind.ExpressionStatement && root.parent.kind !== SyntaxKind.ExpressionStatement;

```

Comment:
Could you just take an `emitParenthesized` parameter instead of this synthetic business?